### ISSUE TITLE
Remove `version` and `name` from package-lock.json

### ISSUE THREAD

**Issue by alexanderby** [npm/npm#19918]
#### What's the feature?
There seems to be no need in version number and package name in `package-lock.json`.

#### What problem is the feature intended to solve?
Version numbers in `package.json` and `package-lock.json` become unsynchronized. There becomes a need to edit this number manually or execute `npm install` before committing incremented package version number when preparing to publish.

#### Is this feature similar to an existing feature in another tool?
`yarn.lock` has no package name or version number.

**Comment by papandreou** [comment]
I agree with this, especially for `version`. For projects that do frequent releases (or have long running feature branches) without necessarily updating dependencies, this is causing `package-lock.json` to be updated with the newest version number and no other changes. If you commit those "non-changes", it's a source of trivial merge conflicts, and if you don't, it'll keep popping up.
